nexus/db-queries/src/db/datastore/silo.rs

```diff
@@ -338,16 +337,15 @@ impl DataStore {
         // Make sure there are no projects present within this silo.
         let id = authz_silo.id();
         let rcgen = db_silo.rcgen;
-        let project_found = diesel_result_optional(
-            project::dsl::project
-                .filter(project::dsl::silo_id.eq(id))
-                .filter(project::dsl::time_deleted.is_null())
-                .select(project::dsl::id)
-                .limit(1)
-                .first_async::<Uuid>(&*conn)
-                .await,
-        )
-        .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+        let project_found = project::dsl::project
+            .filter(project::dsl::silo_id.eq(id))
+            .filter(project::dsl::time_deleted.is_null())
+            .select(project::dsl::id)
+            .limit(1)
+            .first_async::<Uuid>(&*conn)
+            .await
+            .optional()
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
```

Comment:
This is an example of us using https://docs.diesel.rs/master/diesel/prelude/trait.OptionalExtension.html

This works now because we're passing a https://docs.diesel.rs/master/diesel/result/enum.Error.html around, which can go directly to `public_error_from_diesel`.